dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -0,0 +1,450 @@
+        final SiteSearchResults search = siteSearchAPI.search(newIndexName, "*",0, 10);
+        Assert.assertTrue(search.getTotalResults() >= 1);
+        final List<SiteSearchAudit> recentAudits = siteSearchAuditAPI.findRecentAudits(jobId, 0, 1);
+        Assert.assertFalse(recentAudits.isEmpty());
+        final SiteSearchAudit siteSearchAudit = recentAudits.get(0);
+        Assert.assertEquals(1, siteSearchAudit.getPagesCount());
+        Assert.assertFalse(siteSearchAudit.isIncremental()); //did not run incrementally since it was launched as a run-now.
+        Assert.assertTrue(siteSearchAPI.isDefaultIndex(newIndexName));
+    }
+
+
+    @Test
+    public void Test_Non_Incremental_Create_Default_Index_Create_Second_Index_Run_Non_Incrementally_Expect_Non_Default_New_Index()
+            throws DotPublishingException, JobExecutionException, DotDataException, IOException, DotSecurityException {
+
+        List<SiteSearchAudit> recentAudits;
+        final SiteSearchJobImpl impl = new SiteSearchJobImpl();
+
+        final List<String> indicesBeforeTest = siteSearchAPI.listIndices();
+        for(final String index:indicesBeforeTest) {
+            esIndexAPI.delete(index);
+        }
+
+        final long timeMillis1 = System.currentTimeMillis();
+        final String defaultAlias = IndexType.SITE_SEARCH.getPrefix() + "_alias_" + timeMillis1;
+        final String defaultIndexName = IndexType.SITE_SEARCH.getPrefix() + "_" + timeMillis1;
+        siteSearchAPI.createSiteSearchIndex(defaultIndexName, defaultAlias, 1);
+        siteSearchAPI.activateIndex(defaultIndexName); //Make it default.
+
+        final String jobId1 = UUIDUtil.uuid();
+
+        final JobDataMap jobDataMap1 = new JobDataMap();
+        jobDataMap1.put(SiteSearchJobImpl.RUN_NOW, Boolean.TRUE.toString());
+        jobDataMap1.put(SiteSearchJobImpl.INCREMENTAL, Boolean.FALSE.toString());
+        jobDataMap1.put(SiteSearchJobImpl.INDEX_ALIAS, defaultAlias);
+        jobDataMap1.put(SiteSearchJobImpl.JOB_ID, jobId1);
+        jobDataMap1.put(SiteSearchJobImpl.QUARTZ_JOB_NAME, SiteSearchJobImpl.RUNNING_ONCE_JOB_NAME);
+        jobDataMap1.put(SiteSearchJobImpl.INCLUDE_EXCLUDE, "all");
+        jobDataMap1.put(SiteSearchJobImpl.LANG_TO_INDEX, new String[]{Long.toString(defaultLang)});
+        jobDataMap1.put(SiteSearchJobImpl.INDEX_HOST, site.getIdentifier());
+
+        final JobDetail jobDetail1 = Mockito.mock(JobDetail.class);
+        Mockito.when(jobDetail1.getJobDataMap()).thenReturn(jobDataMap1);
+        final JobExecutionContext context1 = Mockito.mock(JobExecutionContext.class);
+        Mockito.when(context1.getJobDetail()).thenReturn(jobDetail1);
+        Mockito.when(context1.getFireTime()).thenReturn(new Date());
+
+        impl.run(context1);
+        //First run will create the default index
+
+        List <String> indices = siteSearchAPI.listIndices();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'indices' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093862)

Reply:
done